Decode/EventHeaderEnumerator.cs

```diff
@@ -1148,25 +1152,13 @@ public EventHeaderItemInfo GetItemInfo()
         public EventHeaderItemInfo GetItemInfo(ReadOnlySpan<byte> eventDataSpan)
         {
             Debug.Assert(m_eventData.Length == eventDataSpan.Length);
-
-            if (m_state <= EventHeaderEnumeratorState.BeforeFirstItem)
-            {
-                throw new InvalidOperationException(); // PRECONDITION
-            }
-
```

Comment:
Note: This precondition is checked in the call to GetItemMetadata, so we don't need to check it here.